src/redux/stockSlice.js

```diff
@@ -0,0 +1,77 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const apiURL = 'https://api.polygon.io/v3/reference/tickers?market=stocks&active=true&apiKey=Vp4PFtmRiJhA03_A8hWFal2787C86E4P';
+
+// fetching stocks form home page to render the list of stocks in home page
+export const fetchData = createAsyncThunk('stocks/fetchData', async () => {
+  const response = await fetch(apiURL);
+  if (response.ok) {
+    const data = await response.json();
+    return data.results;
+  }
+  throw new Error('Failed to fetch data');
+});
+
+// fetching details of stock passing the parameter of stock to the URL
+export const fetchStockInfo = createAsyncThunk(
+  'stocks/fetchStockInfo',
+  async (stock) => {
+    const response = await fetch(`https://api.polygon.io/v3/reference/tickers/${stock}?apiKey=Vp4PFtmRiJhA03_A8hWFal2787C86E4P`);
+    if (response.ok) {
+      const data = await response.json();
+      return data.results;
+    }
+    throw new Error('Failed to fetch stock information');
+  },
+);
+
+const initialState = {
+  status: 'idle',
+  data: [],
+  stockInfo: {},
+
+};
+
+export const stockSlice = createSlice({
+  name: 'stocks',
+  initialState,
+  reducers: {
+    selectStock: (state, action) => {
+      const selectedStock = action.payload;
+      return {
+        ...state,
+        stockInfo: selectedStock,
+      };
+    },
+  },
```

Comment:
- [ ] I still don't on the webpage a way where I can search/filter what I want to see, kindly implement this requirement
![image](https://github.com/danielamoreno699/react-project-capstone/assets/62705947/6baf5e41-91f6-49dd-b290-49f6ae9dc8e6)
![image](https://github.com/danielamoreno699/react-project-capstone/assets/62705947/863229b2-87ce-4c2f-bc18-591212496d6d)